Automattic/wp-calypso
# State: Introduce Jetpack user connection data to Redux

This PR introduces the Redux functionality for retrieving the connection data of the current Jetpack user. It will be necessary in order to create settings blocks, related with managing the user's connection. This PR lays most of the groundwork for #13232.

The PR introduces the following:

* `getJetpackUserConnectionData()` wpcom undocumented method to query the Jetpack site's REST API and fetch the user connection data.
* `requestJetpackUserConnectionData` Redux action to trigger a user connection data request.
* `dataItems` and `dataRequests` reducers to handle user connection data requests and items.
* `isRequestingJetpackUserConnection` selector that checks whether a user connection data request is in progress for a site.
* `getJetpackUserConnection` selector that retrieves the user connection data for a particular site.
* `isJetpackUserMaster` selector that returns whether the current user is the master user that owns the connection.
* `QueryJetpackUserConnection` query component to manage fetching of the user connection data.
* Tests for all of the above.

To test:

* Checkout this branch
* Verify all of the following tests pass:
  * `npm run test-client client/state/jetpack/connection/test/actions.js`
  * `npm run test-client client/state/jetpack/connection/test/reducer.js`
  * `npm run test-client client/state/selectors/`
* Drop the `QueryJetpackUserConnection` component somewhere and verify it performs the expected network requests.
